extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -720,13 +729,55 @@ export class AmpAnalytics extends AMP.BaseElement {
   expandAndSendRequest_(request, trigger, event) {
     this.config_['vars']['requestCount']++;
     const expansionOptions = this.expansionOptions_(event, trigger);
+    /** @type {!Object<string, (DynamicVariableBindingDef|HtmlAttrBindingDef)>} */
     const dynamicBindings =
         this.getDynamicVariableBindings_(trigger, expansionOptions);
+    dynamicBindings['HTML_ATTR'] = this.htmlAttrBinding_.bind(this);
     request.send(
         this.config_['extraUrlParams'], trigger, expansionOptions,
         dynamicBindings);
   }
 
+  /**
+   * Provides a binding for getting attributes from the DOM.
+   * Most such bindings are provided in src/service/url-replacements-impl, but
+   * this one needs access to this.win.document, which if the amp-analytics
+   * tag is contained within an amp-ad tag will NOT be the parent/publisher
+   * page. Hence the need to put it here.
+   * @param {string} cssSelector Elements matching this selector will be
+   *     included, provided they have at least one of the attributeNames
+   *     set, up to a max of 10.
+   * @param attributeNames The attributes whose values will be returned
+   * @returns {string}
+   */
+  htmlAttrBinding_(cssSelector, ...attributeNames) {
```

Comment:
would it be possible to move it to A4AVariableSource.

Reply:
@lannka  I'm working on this, but it would require an exception to the dependency rules. Is that ok?